### ISSUE TITLE
Redshift exclude clause throws unparseable error when adding other columns

### ISSUE THREAD

**Issue by holly-evans** [sqlfluff/sqlfluff#7899]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff handles Redshift's exclude clause when selecting `*`.

It fails when you select more than `*`, which is valid for Redshift.

```sql
SELECT *, NULL AS example EXCLUDE (col1, col2) FROM table1;
```

```
╰─➤  sqlfluff fix                                                                                                                       1 ↵
==== finding fixable violations ====
  [1 templating/parsing errors found]                                                                                                       
  == [/Users/hevans/code/python/sqlfluff_bug/select_star_plus.sql] ==                                                                       
  L:   1 | P:  27 |  PRS | Line 1, Position 27: Found unparsable section: 'EXCLUDE
                       | (col1, col2)'
==== no fixable linting violations found ====
All Finished 📜 🎉!
```

### Expected Behaviour

No parsing error.

### Observed Behaviour

Parsing error.

### How to reproduce

.sql
```sql
SELECT *, NULL AS example EXCLUDE (col1, col2) FROM table1;
```

.sqlfluff
```
[sqlfluff]
dialect = redshift
rules = core
```

`sqlfluff fix`

### Dialect

Redshift

### Version

```
╰─➤  sqlfluff --version                                                                                                                 1 ↵
sqlfluff, version 4.2.1
```

### Configuration

```
[sqlfluff]
dialect = redshift
rules = core
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @holly-evans  - thanks for opening this issue.

I have looked at the example, and I could not reconcile this with the [official Redshift EXCLUDE syntax](https://docs.aws.amazon.com/redshift/latest/dg/r_EXCLUDE_list.html)

I think the following is a valid SQLs:
```sql
SELECT * EXCLUDE (col1, col2), NULL AS example FROM table1;
```

while your example (`SELECT *, NULL AS example EXCLUDE (col1, col2) FROM table1;`) seems to me syntactically incorrect, as the EXCLUDE is NOT following the `*` selector.

**Comment by holly-evans** [comment]
@peterbud Do you have access to Redshift to try it? This syntax is valid in my tests.

The `EXCLUDE` documentation does not say it has to follow the `*` selector, it just does not have examples of more complicated queries. Looking at the documentation one level up for the select statement (https://docs.aws.amazon.com/redshift/latest/dg/r_SELECT_synopsis.html), it shows `EXCLUDE` coming after the entire select list, so the `EXCLUDE` must be at the end.

Here are some tests on a small table in my cluster.

`SELECT top 5 * FROM dbo.inactive_client_status;`
<img width="828" height="314" alt="Image" src="https://github.com/user-attachments/assets/b4206d78-4de8-4f74-bc1d-303e1bdfd4ba" />

`SELECT top 5 *, NULL as example FROM dbo.inactive_client_status;`
<img width="869" height="316" alt="Image" src="https://github.com/user-attachments/assets/cfddd061-dc8e-4b41-a21c-47b760a72613" />

`SELECT top 5 * EXCLUDE (batch_id, batch_date) FROM dbo.inactive_client_status;`
<img width="700" height="315" alt="Image" src="https://github.com/user-attachments/assets/79cf7dfc-5c4d-4c53-8810-0f87e0f2920a" />

`SELECT top 5 *, NULL as example EXCLUDE (batch_id, batch_date) FROM dbo.inactive_client_status;`
<img width="831" height="310" alt="Image" src="https://github.com/user-attachments/assets/032bd72f-985c-4752-b2d3-21b7e4040fc3" />

`SELECT top 5 * EXCLUDE (batch_id, batch_date), NULL as example FROM dbo.inactive_client_status;`
<img width="831" height="87" alt="Image" src="https://github.com/user-attachments/assets/9218c4c1-13c6-4185-a7b3-5c6db96324a7" />

**Comment by peterbud** [comment]
Thanks @holly-evans  - I see now. And I don't have access to Redshift.

In the current dialect grammar the `ExcludeClauseSegment` was added as part of the wildcard expression in #6838, but it seems that this [syntax](https://docs.aws.amazon.com/redshift/latest/dg/r_SELECT_synopsis.html) requires a different approach.

I think we have everything to make a fix.
